client-lib/src/main/java/com/ibm/ws/repository/strategies/writeable/AddThenHideOldStrategy.java

```
@@ -58,106 +53,56 @@ public AddThenHideOldStrategy(State desiredStateIfMatchingFound, State desiredSt
+            // Only hide visible, published resources, and don't hide the one that's just been created!
+            if (!isVisible(resource)) {
+                continue;
             }
-            if (!oldResourceDeleted) {
-                // Since resource object is the resource that is being replaced, we need to get a copy of the
-                // resource before it gets updated and we can then delete that copy (the 4th parameter
-                // passed to the AdddThenDeleteStrategy constructor). There is a chance that the resource will
-                // have been deleted since we setup the cache so do this in a try catch.
-                RepositoryResource resourceToDelete;
-                try {
-                    resourceToDelete = resource.getRepositoryConnection().getResource(resource.getId());
-                } catch (RepositoryException e) {
-                    // Refresh the cache
-                    synchronized (allResources) {
-                        refreshCache(resource.getRepositoryConnection(), newResource.getId());
-                        resource = allResources.get(vanityURL);
-                        if (!(resource instanceof WebDisplayable)) {
-                            // Nothing matches any more so exit
-                            return;
-                        }
-                        resourceToDelete = resource.getRepositoryConnection().getResource(resource.getId());
-                    }
-                }
-                ((WebDisplayable) resource).setWebDisplayPolicy(DisplayPolicy.HIDDEN);
-                resource.uploadToMassive(new AddThenDeleteStrategy(null, State.DRAFT, true, (RepositoryResourceImpl) resourceToDelete));
+            if (!State.PUBLISHED.equals(((RepositoryResourceWritable) resource).getState())) {
+                continue;
             }
+            if (newResource.getId().equals(resource.getId())) {
+                continue;
+            }
+
+            if (found) {
```

I'm still not clear whether or not we want to do this check.

With the logic as-is, we'll hide one of the visible resources, and then throw the exception.

If we really want to throw an exception here, I think we should be doing all the checking before we perform any actions.